src/GafferScene/PrimitiveSampler.cpp

```diff
@@ -0,0 +1,385 @@
+		input == sourcePlug() ||
+		input == sourceLocationPlug() ||
+		input == primitiveVariablesPlug() ||
+		input == prefixPlug() ||
+		input == statusPlug() ||
+		input == sourcePlug()->objectPlug() ||
+		input == inPlug()->transformPlug() ||
+		input == sourcePlug()->transformPlug() ||
+		affectsSamplingFunction( input )
+	;
+}
+
+void PrimitiveSampler::hashProcessedObject( const ScenePath &path, const Gaffer::Context *context, IECore::MurmurHash &h ) const
+{
+	PrimitiveSampler::Deformer::hashProcessedObject( path, context, h );
+
+	const string sourceLocation = sourceLocationPlug()->getValue();
+	const string primitiveVariables = primitiveVariablesPlug()->getValue();
+	const string status = statusPlug()->getValue();
+	if( sourceLocation.empty() || ( primitiveVariables.empty() && status.empty() ) )
+	{
+		return;
+	}
+
+	ScenePlug::ScenePath sourcePath;
+	ScenePlug::stringToPath( sourceLocation, sourcePath );
+	if( !SceneAlgo::exists( sourcePlug(), sourcePath ) )
+	{
+		return;
+	}
+
+	h.append( sourcePlug()->objectHash( sourcePath ) );
+	h.append( primitiveVariables );
+	prefixPlug()->hash( h );
+	h.append( status );
+	h.append( inPlug()->fullTransformHash( path ) );
+	h.append( sourcePlug()->fullTransformHash( sourcePath ) );
+	hashSamplingFunction( h );
+}
+
+IECore::ConstObjectPtr PrimitiveSampler::computeProcessedObject( const ScenePath &path, const Gaffer::Context *context, const IECore::Object *inputObject ) const
+{
+	const Primitive *inputPrimitive = runTimeCast<const Primitive>( inputObject );
+	if( !inputPrimitive )
+	{
+		return inputObject;
+	}
+
+	const string sourceLocation = sourceLocationPlug()->getValue();
+	const string primitiveVariables = primitiveVariablesPlug()->getValue();
+	const string status = statusPlug()->getValue();
+	if( sourceLocation.empty() || ( primitiveVariables.empty() && status.empty() ) )
+	{
+		return inputObject;
+	}
+
+	PrimitiveVariable::Interpolation outputInterpolation = PrimitiveVariable::Invalid;
+	SamplingFunction samplingFunction = computeSamplingFunction( inputPrimitive, outputInterpolation );
+	if( !samplingFunction || outputInterpolation == PrimitiveVariable::Invalid )
+	{
+		return inputObject;
+	}
+
+	ScenePlug::ScenePath sourcePath;
+	ScenePlug::stringToPath( sourceLocation, sourcePath );
+	if( !SceneAlgo::exists( sourcePlug(), sourcePath ) )
+	{
+		return inputObject;
+	}
+
+	ConstObjectPtr sourceObject = sourcePlug()->object( sourcePath );
+	const Primitive *sourcePrimitive = runTimeCast<const Primitive>( sourceObject.get() );
+	if( !sourcePrimitive )
+	{
+		return inputObject;
+	}
+
+	ConstPrimitivePtr preprocessedSourcePrimitive = sourcePrimitive;
+	if( auto mesh = runTimeCast<const MeshPrimitive>( preprocessedSourcePrimitive.get() ) )
+	{
+		preprocessedSourcePrimitive = MeshAlgo::triangulate( mesh );
+	}
+	PrimitiveEvaluatorPtr evaluator = PrimitiveEvaluator::create( preprocessedSourcePrimitive );
+	if( !evaluator )
+	{
+		return inputObject;
```

Comment:
Also feels like an error case?

Reply:
See earlier comments. I'll message you separately with an anecdote that might also convince you here :)

Reply:
Also bear in mind that we have evaluators for points/curves/meshes - I don't think any other primitive types are actually used in Gaffer?